update: readme changes for 9000 upgrade (#761)
ava-labs/coreth: precompile/contracts/warp/README.md

Therefore, we use the [Predicate Utils](https://github.com/ava-labs/coreth/blob/master/predicate/Predicate.md) package to encode the actual byte slice of size N into the access list.

### Performance Optimization: C-Chain to Avalanche L1

> For communication between the C-Chain and L1, as well as broader interactions between the Primary Network and Avalanche L1s, we've implemented special handling for the C-Chain. This special case is necessary since the Primary Network operates with the maximum number of possible validators.

Reviewer: ```suggestion
For communication between the C-Chain and L1, as well as broader interactions between the Primary Network and Avalanche L1s, we've implemented special handling for the C-Chain.
```

> This large validator set in the Primary Network creates a unique challenge for Avalanche Warp Messages. Since reaching the required stake threshold would demand collecting numerous signatures, verifying messages from the Primary Network would be computationally intensive and costly. However, we've developed a more efficient solution.

Reviewer: ```suggestion
The Primary Network has a large validator set, which creates a unique challenge for Avalanche Warp Messages: Since reaching the required stake threshold would demand collecting numerous signatures, and verifying messages from the Primary Network would be computationally costly. However, we've developed a more efficient solution.
```

> When an Avalanche L1 receives a message from a blockchain on the Primary Network, we use the validator set of the receiving L1 instead of the entire network when validating the message. This means that the C-Chain sending a message can be the exact same as Subnet to Subnet communication.

Reviewer: ```suggestion
When an Avalanche L1 receives a message from a blockchain on the Primary Network, we use the validator set of the receiving L1 instead of the entire network when validating the message. Note this is NOT possible if an L1 does not validate the Primary Network, in which case the Warp precompile must be configured with `requirePrimaryNetworkSigners`.
```

> However, when L1 XYZ receives a message from the C-Chain, it changes the semantics to the following:

Reviewer: ```suggestion
Sending messages from the C-Chain remains unchanged.
However, when L1 XYZ receives a message from the C-Chain, it changes the semantics to the following:
```

Therefore, we use the [Predicate Utils](https://github.com/ava-labs/coreth/blob/master/predicate/Predicate.md) package to encode the actual byte slice of size N into the access list.

### Performance Optimization: C-Chain to Avalanche L1

> For communication between the C-Chain and L1, as well as broader interactions between the Primary Network and Avalanche L1s, we've implemented special handling for the C-Chain.

Reviewer 2: minor improvements:

```suggestion
For communication between the C-Chain and an L1, as well as broader interactions between the Primary Network and Avalanche L1s, we have implemented special handling for the C-Chain.
```

> The Primary Network has a large validator set, which creates a unique challenge for Avalanche Warp Messages: Since reaching the required stake threshold would demand collecting numerous signatures, and verifying messages from the Primary Network would be computationally costly. However, we've developed a more efficient solution.

Reviewer 2: - don't use uppercase for common words i.e. `Since`
- rephrased the 2 reasons a bit since it didn't make too much sense grammatically - feel free to re-adjust in case that's not the desired meaning.

```suggestion
The Primary Network has a large validator set, which creates a unique challenge for Avalanche Warp messages. To reach the required stake threshold, numerous signatures would need to be collected and verifying messages from the Primary Network would be computationally costly. However, we have developed a more efficient solution.
```
